include/uv.h

```diff
@@ -1414,9 +1417,35 @@ UV_EXTERN void uv_key_delete(uv_key_t* key);
 UV_EXTERN void* uv_key_get(uv_key_t* key);
 UV_EXTERN void uv_key_set(uv_key_t* key, void* value);
 
+
+/*
+ * CPU masks for thread affinity must be at least UV_CPU_SETSIZE bytes long.
+ */
+#if defined(_WIN32)
+# if defined(_WIN64)
+#  define UV_CPU_SETSIZE        64
+# else
+#  define UV_CPU_SETSIZE        32
+# endif
+#else
+# if (defined(__APPLE__) && defined(__MACH__)) || defined(_AIX)
+#  define UV_CPU_SETSIZE        8
+# else
+#  define UV_CPU_SETSIZE        CPU_SETSIZE
+# endif
+#endif
+
 typedef void (*uv_thread_cb)(void* arg);
 
 UV_EXTERN int uv_thread_create(uv_thread_t* tid, uv_thread_cb entry, void* arg);
+UV_EXTERN int uv_thread_setaffinity(uv_thread_t* tid,
+                                    char *cpumask,
+                                    char *oldmask,
+                                    size_t mask_size);
```

Comment:
``SetThreadAffinityMask()`` returns it, so rather than discard the information, I added getting the previous mask to the UNIX implementation. I'm happy to remove it; should I?